targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/mbed_crc_api.c

```diff
@@ -0,0 +1,125 @@
+#include "crc_api.h"
+
+#include "drivers/fsl_crc.h"
+#include "platform/mbed_assert.h"
+
+#ifdef DEVICE_CRC
+
+static crc_bits_t width;
+
+bool hal_crc_is_supported(const uint32_t polynomial)
+{
+  switch (polynomial)
+  {
+    case POLY_8BIT_CCITT:
+    case POLY_7BIT_SD:
+    case POLY_16BIT_CCITT:
+    case POLY_16BIT_IBM:
+    case POLY_32BIT_ANSI:
+      return true;
+    default:
+      return false;
+  }
+}
+
+void hal_crc_compute_partial_start(const uint32_t polynomial)
+{
+  crc_config_t config;
+
+  switch (polynomial)
+  {
+    case POLY_32BIT_ANSI:
+    {
+      width = kCrcBits32;
+
+      config.polynomial         = polynomial;
+      config.seed               = 0xFFFFFFFFU;
+      config.reflectIn          = true;
+      config.reflectOut         = true;
+      config.complementChecksum = true;
+      config.crcBits            = width;
+      config.crcResult          = kCrcFinalChecksum;
+
+      break;
+    }
+    case POLY_16BIT_IBM:
+    {
+      width = kCrcBits16;
+
+      config.polynomial         = polynomial;
+      config.seed               = 0;
+      config.reflectIn          = true;
+      config.reflectOut         = true;
+      config.complementChecksum = false;
+      config.crcBits            = width;
+      config.crcResult          = kCrcFinalChecksum;
+
+      break;
+    }
+    case POLY_16BIT_CCITT:
+    {
+      width = kCrcBits16;
+
+      config.polynomial         = polynomial;
+      config.seed               = 0xFFFFFFFFU;
+      config.reflectIn          = false;
+      config.reflectOut         = false;
+      config.complementChecksum = false;
+      config.crcBits            = width;
+      config.crcResult          = kCrcFinalChecksum;
+
+      break;
+    }
+    case POLY_8BIT_CCITT:
+    {
+      width = kCrcBits16;
```

Comment:
Why `width` is set to 16-bits if 8-bit CRC is requested?

Reply:
There are only two options, 16 and 32 bit, they correspond to the hardware returning the data from the HI-LO data registers for 32 bit, or LO data register for 16 bit. Any CRC's less than 16 bits will be returned in the LO register only as a 16 bit value.

Reply:
If hardware can handle only 16 or 32 bits CRC, then `POLY_8BIT_CCITT`, `POLY_7BIT_SD` polynomials should be not supported by this platform.

Reply:
They are supported. The result of the CRC calculation is stored in two 16 bit data registers in the CRC module. For CRCs less than 16-bits, only the 16-bit register is used, the other register may contain junk. Setting the width to 16-bit will return only the 16-bit register, which will contain the 7/8 bit result of the calculation.

Reply:
Maybe I'm missing something, but according to K64F reference manual:

> The cyclic redundancy check (CRC) module generates 16/32-bit CRC code for error
> detection.

It looks like currently `POLY_8BIT_CCITT` and `POLY_7BIT_SD` generates 16 bit CRC. We expect to have 8/7 bit CRC. `config.crcBits` is set in both cases to `kCrcBits16` (and on K64F platform we don't have definitions for 7/8 bits CRC). Only different polynomials are used in each case.

Please look at the following test:
```
    uint32_t crc = 0;
    const uint8_t test_data[] = "123456789";
    hal_crc_compute_partial_start(POLY_8BIT_CCITT);
    hal_crc_compute_partial((uint8_t*) test_data, strlen((const char*)test_data));
    crc = hal_crc_get_result();

    printf("Result: 0x%X \n", crc);
    printf("Expected: 0x%X \n", 0xF4);

output:
Result: 0xEF6F  <--- 16 bit result
Expected: 0xF4

I took expected result (0xF4) from `tests-mbed_drivers-crc`.
```

Reply:
Ah, you're correct, I'm wrong, I thought there was another option for the polynomial width. I'll remove them from the code.